core/src/main/java/org/infinispan/interceptors/impl/TransactionalExceptionEvictionInterceptor.java

```diff
@@ -166,6 +168,31 @@ public Object visitInvalidateCommand(InvocationContext ctx, InvalidateCommand co
       return super.visitInvalidateCommand(ctx, command);
    }
 
+   // Remove Expired is not transactional
+   @Override
+   public Object visitRemoveExpiredCommand(InvocationContext ctx, RemoveExpiredCommand command) {
+      Object key = command.getKey();
+      // Skip adding changeAmount if originator is not primary
+      if (ctx.isOriginLocal() && dm != null && !dm.getCacheTopology().getSegmentDistribution(command.getSegment()).isPrimary()) {
+         return invokeNext(ctx, command);
+      }
+      return invokeNextThenAccept(ctx, command, ((rCtx, rCommand, rv) -> {
+         if (rCommand.isSuccessful()) {
```

Comment:
lambda can be cached (sound like Dan xD)

Reply:
Sure.